pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -536,6 +514,31 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 					newMapperErr)
 			}
 		}
+	} else {
+		var err error
+		volumeMounter, err = plugin.NewMounter(
+			volumeSpec,
+			pod,
+			volumepkg.VolumeOptions{})
+		if err != nil {
+			return nil, fmt.Errorf(
+				"reconstructVolume.NewMounter failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v",
+				uniqueVolumeName,
+				volumeSpec.Name(),
+				volume.podName,
+				pod.UID,
+				err)
+		}
+	}
+
+	// Check existence of mount point for filesystem volume or symbolic link for block volume
+	isExist, checkErr := rc.operationExecutor.CheckVolumeExistenceOperation(volumeSpec, volume.volumePath, volumeSpec.Name(), rc.mounter, uniqueVolumeName, volume.podName, pod.UID, attachablePlugin)
```

Comment:
`volume.volumePath` is used incorrectly here. Before this PR, `volumeMounter.GetPath()` was used. This breaks CSI reconstruction, as `CheckVolumeExistenceOperation` checks the pod directory [1] and not the directory where the volume should be mounted [2]

1: `/var/lib/kubelet/pods/4ec8e3aa-f83e-4d49-96ea-3859f09f7dd7/volumes/kubernetes.io~csi/pvc-873e54a8-a2c1-4872-8a3c-b68b71a81e42`
2: `/var/lib/kubelet/pods/4ec8e3aa-f83e-4d49-96ea-3859f09f7dd7/volumes/kubernetes.io~csi/pvc-873e54a8-a2c1-4872-8a3c-b68b71a81e42/mount`